### PR TITLE
Resolve regressions in Posts Controller

### LINKED ISSUE #738: Regressions in post controller
#684 introduced some regressions in the post controller from v1.

* `setup_postdata` is not called before preparing the post, which breaks filtering (such as `the_content`)
* Passworded posts are not handled correctly
* `post_date` is checked for `0000-00-00 00:00:00`, but `post_date_gmt` should be checked instead

### PR DESCRIPTION
**Pull request by rachelbaker**
Fixes #738

### CODE REVIEW

**Comment by rachelbaker** [comment]
@rmccue #reviewmerge

**Comment by danielbachhuber** [comment]
@rachelbaker  should we add tests for the other bugs?

File: tests/test-json-posts-controller.php [hunk 1]
```diff
@@ -148,6 +148,16 @@ public function test_get_post_context_without_permisson() {
...
+	public function test_get_post_with_password_without_permisson() {
+		$post_id = $this->factory->post->create( array(
+			'post_password' => 'always$inthebananastand',
+		) );
+		$request = new WP_JSON_Request( 'GET', sprintf( '/wp/posts/%d', $post_id ) );
+		$response = $this->server->dispatch( $request );
+
+		$this->assertErrorResponse( 'json_user_cannot_read', $response, 401 );
+	}
+
```
Comment (danielbachhuber) [inline tests/test-json-posts-controller.php]: We should have a test for with permission too

Comment (rachelbaker) [inline tests/test-json-posts-controller.php]: @danielbachhuber We actually already had this covered in the `test_create_post_with_password()` method.

**Comment by rachelbaker** [comment]
We haz unit tests

**Pull request merged**